tensorflow_quantum/python/optimizers/rotosolve_minimizer.py

```diff
@@ -0,0 +1,284 @@
+# Copyright 2020 The TensorFlow Quantum Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""The rotosolve minimization algorithm
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+import math
+import collections
+import tensorflow as tf
+
+
+def prefer_static_shape(x):
+    """Return static shape of tensor `x` if available,
+     else `tf.shape(x)`.
+    Args:
+      x: `Tensor` (already converted).
+    Returns:
+      Numpy array (if static shape is obtainable), else `Tensor`.
+    """

```

Comment:
we are using "four space" indent in comments, with a one-line description header with period.
e.g.
```python
"""Static shape of a given tensor `x`.

Return static shape of tensor `x` if available, else `tf.shape(x)`.

Args:
    x: A `tf.Tensor` of which we want to know the shape.
Returns:
    A numpy array (if static shape is obtainable), else `Tensor`
    containing a shape of `x`.
"""
```

Reply:
Thanks! Fixed in 112400e847d6d98588c1708517aed5b7b7aeeb43